schema/testing.proto

```diff
@@ -238,4 +230,128 @@ message Test {
+    string id = 1;
+
+    // A description of this blueprint.
+    string description = 2;
+
+    // A message representing a method invocation.
+    message Invocation {
+      // The fully qualified name of the showcase method to be invoked.
+      string method = 1;
+
+      // The request to be made if a specific request is necessary.
+      google.protobuf.Any request = 2;
+    }
+
+    // An ordered list of method calls that can be called to trigger this test.
+    repeated Invocation invocations = 3;
+  }
+
+  // The blueprints that will satisfy this test. There may be multiple blueprints
+  // that can signal to the server that this test case is being exercised. Although
+  // multiple blueprints are specified, only a single blueprint needs to be run to
+  // signal that the test case was exercised.
+  repeated Blueprint blueprints = 4;
+}
+
+// An issue found in the test.
+message Issue {
+  // The different potential types of issues.
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+
+    // The test was never instrumented.
+    SKIPPED = 1;
+
+    // The test was started but never confirmed.
+    PENDING = 2;
+
+    // The test was instrumented, but Showcase got an unexpected
+    // value when the generator tried to confirm success.
+    INCORRECT_CONFIRMATION = 3;
+  }
+
+  // The type of the issue.
+  Type type = 1;
+
+  // Severity levels.
+  enum Severity {
+    SEVERITY_UNSPECIFIED = 0;
+
+    // Errors.
+    ERROR = 1;
+
+    // Warnings.
+    WARNING = 2;
+  }
+
+  // The severity of the issue.
+  Severity severity = 2;
+
+  // A description of the issue.
+  string description = 3;
+}
+
+// The request for the ListTests method.
+message ListTestsRequest {
+  // The session.
+  string parent = 1 [(google.api.resource_reference) = "Session"];
+
+  // The maximum number of tests to return per page.
+  int32 page_size = 2;
+
+  // The page token, for retrieving subsequent pages.
+  string page_token = 3;
+}
+
+// The response for the ListTests method.
+message ListTestsResponse {
+  // The tests being returned.
+  repeated Test tests = 1;
+
+  // The next page token, if any.
+  // An empty value here means the last page has been reached.
+  string next_page_token = 2;
+}
+
+// A TestRun is the result of running a Test.
+message TestRun {
+  // The name of the test.
+  // The tests/* portion of the names are hard-coded, and do not change
+  // from session to session.
+  string test = 1 [(google.api.resource_reference) = "Test"];
+
+
+  // An issue found with the test run. If empty, this test run was successful.
+  Issue issue = 2;
+}
+
+// Request message for deleting a test.
+message DeleteTestRequest {
+  // The test to be deleted.
+  string name = 1 [(google.api.resource_reference) = "Test"];
+}
+
+message CheckAnswerRequest {
+  // The test to have an answer registered to it.
+  string name = 1 [(google.api.resource_reference) = "Test"];
+
+  // The answer from the test.
+  google.protobuf.Any answer = 2;
```

Comment:
Are you able to know what type should be sent back? If so, using a `bytes` here is likely to be better.

Reply:
Done!